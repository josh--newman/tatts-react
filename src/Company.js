import React from 'react';
import PropTypes from 'prop-types';

const Company = ({ logoUrl, description }) => {
  return (
    <li className="list-item">
      <div className="logo">
        <img src={logoUrl} alt="logo" />
      </div>
      <p>{description}</p>
      <span className="arrow">></span>
    </li>
  );
};

Company.propTypes = {
  logoUrl: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Company;
