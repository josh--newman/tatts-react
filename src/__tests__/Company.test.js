import React from 'react';
import { shallow } from 'enzyme';
import Company from '../Company';

const baseProps = {
  logoUrl: 'https://example.com/api/stuff.png',
  description: 'Something'
}

describe('Company', function() {
  it('renders correctly', function() {
    expect(shallow(<Company {...baseProps} />))
      .toMatchSnapshot();
  });
});
