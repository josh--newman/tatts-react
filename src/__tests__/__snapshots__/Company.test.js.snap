// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Company
    description="Something"
    logoUrl="https://example.com/api/stuff.png"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo"
>
          <img
                    alt="logo"
                    src="https://example.com/api/stuff.png"
          />
</div>,
        <p>
          Something
</p>,
        <span
          className="arrow"
>
          &gt;
</span>,
      ],
      "className": "list-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo"
            src="https://example.com/api/stuff.png"
/>,
          "className": "logo",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "src": "https://example.com/api/stuff.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Something",
        },
        "ref": null,
        "rendered": "Something",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": ">",
          "className": "arrow",
        },
        "ref": null,
        "rendered": ">",
        "type": "span",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo"
>
            <img
                        alt="logo"
                        src="https://example.com/api/stuff.png"
            />
</div>,
          <p>
            Something
</p>,
          <span
            className="arrow"
>
            &gt;
</span>,
        ],
        "className": "list-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              src="https://example.com/api/stuff.png"
/>,
            "className": "logo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "src": "https://example.com/api/stuff.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Something",
          },
          "ref": null,
          "rendered": "Something",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": ">",
            "className": "arrow",
          },
          "ref": null,
          "rendered": ">",
          "type": "span",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
